### ISSUE TITLE
Explainer feedback

### ISSUE THREAD

**Issue by jakearchibald** [WICG/file-handling#20]
>  If the web application accepts all text and image formats, "text/*" and "image/*" could be used

SIde note, but `image/*` feels kinda misleading, as it's restricted to types the browser supports, not all types that are images.

---

> // Cause of the launch (e.g. file_handler|share_target|shortcut|link). Only files will be supported initially but will likely be added in future. This key would be based on the manifest entry, where appropriate.

Nit: We generally use hyphens rather than underscores. Except in the manifest spec [along sigh].

"This key would be based on the manifest entry" I'm not sure what this means. The start of the sentence makes it sound like an enum.

---

It isn't clear how this integrates with the launch event.

---

Is there a reason you're using an event to deliver the files rather than a property accessible from the global? Is it a GC thing?

---

```js
if (!client)
  clients.openWindow(url);
  return;
}
```

Nit: should be `client = await clients.openWindow(url)`.

**Comment by fallaciousreasoning** [comment]
> SIde note, but image/* feels kinda misleading, as it's restricted to types the browser supports, not all types that are images.

I'll update the explainer to note that :)

---

> Nit: We generally use hyphens rather than underscores. Except in the manifest spec [along sigh].

Yeah, that's where we're planning on getting the keys from :/ For example, the 'file_handler' entry in the manifest means we'll use 'file_handler' as the key in the cause enum. I'll reword this to be clearer.

---

> Is there a reason you're using an event to deliver the files rather than a property accessible from the global? Is it a GC thing?

I think we're considering moving towards a `getLaunchParams() => Promise<LaunchParams>` kind of approach. There's some concern that the events wouldn't fire at the right point in time, and the restriction of 'You can only get this in the event' seems kind of strange.

---

> Nit: should be `client = await clients.openWindow(url)`.

Nice catch! :)

---

PR #21

**Comment by raymeskhoury** [comment]
Thanks Jake! 

> It isn't clear how this integrates with the launch event.

Do you happen to have any more specifics about what's not clear? There's a small paragraph on it:

> This API is sufficient to allow native files to be opened in web applications. However, it doesn't cater for more advanced use cases, such as opening a file in an existing window, or simply displaying a notification when a file is opened. For these cases applications can add a launch event handler. The launchParams object would be attached to the launch event in an identical manner to the load event.

And an example. Should it be fleshed out more?

**Comment by jakearchibald** [comment]
@raymeskhoury I deleted "It isn't clear how this integrates with the launch event." a couple of minutes after posting.

**Comment by raymeskhoury** [comment]
Ohh cool - sorry I was reading from my email!
